va/va_prot.h

```diff
@@ -0,0 +1,457 @@
+ * while representing protection status in GPU and Display.
+ *
+ * Both contexts and protected sessions are identified by unique IDs and its
+ * implementation specific internals are kept opaque to the clients
+ */
+typedef VAGenericID VAProtectedSessionID;
+
+/** \brief TEE Execucte Function ID. */
+typedef enum _VA_TEE_EXEC_FUNCTION_ID {
+  VA_TEE_EXECUTE_FUNCTION_ID_PASS_THROUGH = 0x00000001,
+  VA_TEE_EXECUTE_FUNCTION_ID_GET_FIRMWARE_VERSION = 0x00000002,
+
+} VA_TEE_EXECUTE_FUNCTION_ID;
+
+/** \brief Input/Output buffer of VAProtectedSessionExecuteBuffer */
+typedef struct _VAProtectedSessionBuffer {
+  /*
+   * This is used when this buffer refer to output buffer. The maximum size of
+   * data that the driver can return in the output buffer. It is not used for
+   * input buffer.
+   */
+  uint32_t max_data_size;
+  /*
+   * If it is used for input buffer, it is the size of the input data. If it is
+   * used for output buffer, it is the returns size of the output data written
+   * by the driver.
+   */
+  uint32_t data_size;
+  /*
+   * data pointer of this buffer
+   */
+  void *data;
+  uint32_t va_reserved[VA_PADDING_LOW];
+} VAProtectedSessionBuffer;
+
+/** \brief Buffer for vaProtectedSessionExecute() */
+typedef struct _VAProtectedSessionExecuteBuffer {
+  /** \brief Specify the function to execute. It is IHV's implementation
+   * specific */
+  uint32_t function_id;
+  /** \brief Input buffer */
+  VAProtectedSessionBuffer input;
+  /** \brief Output buffer */
+  VAProtectedSessionBuffer output;
+  /** \brief Return the result of this function. The status result is IHV's
+   * implementation specific */
+  uint32_t status;
+  uint32_t va_reserved[VA_PADDING_LOW];
+} VAProtectedSessionExecuteBuffer;
+
+/**
+ * \brief Create a protected session
+ *
+ * Create a protected session
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] config_id             configuration for the protected session
+ * @param[out] protected_session    created protected session id upon return
+ */
+VAStatus vaCreateProtectedSession(VADisplay dpy, VAConfigID config_id,
+                                  VAProtectedSessionID *protected_session);
+
+/**
+ * \brief Destroy a protected session
+ *
+ * Destroy a protected session
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] protected_session     protected session to be destroyed
+ */
+VAStatus vaDestroyProtectedSession(VADisplay dpy,
+                                   VAProtectedSessionID protected_session);
+
+/**
+ * \brief Attach a protected content session to VA context
+ *
+ * Attach a protected content session to the context to enable
+ * decryption/protection
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] id                    the VA decode/encode/vp context
+ * @param[in] protected_session     the protected session to attach
+ */
+VAStatus vaAttachProtectedSession(VADisplay dpy, VAGenericID id,
```

Comment:
Why `VAGenericID id` instead of `VAContextID`? what else it can be? VAMFContextID (potentially)?

Reply:
Yes, potentially and possibly to represent other security accelerators.